perf-tests/clusterloader2/pkg/measurement/util/object_store.go

```diff
@@ -74,27 +80,44 @@ func (s *ObjectStore) Stop() {
 // PodStore is a convenient wrapper around cache.Store.
 type PodStore struct {
 	*ObjectStore
+	listener int
 }
 
+var podStore *PodStore = nil
+var initPodStoreLock sync.Mutex
+
 // NewPodStore creates PodStore based on given object selector.
 func NewPodStore(c clientset.Interface, selector *ObjectSelector) (*PodStore, error) {
+	initPodStoreLock.Lock()
+	defer initPodStoreLock.Unlock()
+
+	if podStore != nil {
+		podStore.listener++
+		klog.Infof("0==== ADD PodStore listener, total %v", podStore.listener)
+		return podStore, nil
+	}
+
+	var err error
+	podStore, err = initPodStore(c)
+	return podStore, err
+}
+
+func initPodStore(c clientset.Interface) (*PodStore, error) {
+	klog.Infof("0==== initPodStore")
+	//debug.PrintStack()
 	lw := &cache.ListWatch{
 		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-			options.LabelSelector = selector.LabelSelector
-			options.FieldSelector = selector.FieldSelector
-			return c.CoreV1().PodsWithMultiTenancy(selector.Namespace, util.GetTenant()).List(options)
+			return c.CoreV1().PodsWithMultiTenancy(metav1.NamespaceAll, util.GetTenant()).List(options)
```

Comment:
do we still need the options for the list and watch function? or just use group=latency or group=saturation ?

Reply:
Currently we still have group=latency and group=saturation in the list/watch argument. It might come from options. Leave the option there until it is verified.